packages/editor/src/lib/components/Shape.tsx

```diff
@@ -59,6 +57,27 @@ export const Shape = memo(function Shape({
 			const shape = editor.getShape(id)
 			if (!shape) return // probably the shape was just deleted
 
+			{
+				// If renderingBoundsMargin is set to Infinity, then we won't cull offscreen shapes
+				const isCullingOffScreenShapes = Number.isFinite(editor.renderingBoundsMargin)
+				const selectedShapeIds = editor.getSelectedShapeIds()
+				const renderingBoundsExpanded = editor.getRenderingBoundsExpanded()
+			const maskedPageBounds = editor.getShapeMaskedPageBounds(id)
+				const isCulled =
+					isCullingOffScreenShapes &&
+					// never cull editing shapes
+					editor.getEditingShapeId() !== id &&
+					// if the shape is fully outside of its parent's clipping bounds...
+					(maskedPageBounds === undefined ||
+						// ...or if the shape is outside of the expanded viewport bounds...
+						(!renderingBoundsExpanded.includes(maskedPageBounds) &&
+							// ...and if it's not selected... then cull it
+							!selectedShapeIds.includes(id)))
```

Comment:
This would be the move from isCulled from out of the rendering shapes array and into the shape, right? I saw that on the other (webgl) PR.

Reply:
yep